pkg/verify/signed_entity.go

```diff
@@ -515,6 +515,18 @@ func (v *SignedEntityVerifier) Verify(entity SignedEntity, pb PolicyBuilder) (*V
 		return nil, fmt.Errorf("failed to build policy: %w", err)
 	}
 
+	// Early check: If SCTs are required, ensure the bundle is certificate-signed not public key-signed
+	if v.config.requireSCTs {
+		verificationContent, err := entity.VerificationContent()
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch verification content: %w", err)
+		}
+
+		if leafCert := verificationContent.Certificate(); leafCert == nil {
```

Comment:
Can we check for `.PublicKey()` rather than not `Certificate()`, since we also may support certificate chains at some point?